src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const rezult = [];
+
+  for (const i of actions) {
+    let item = {};
+
+    if (rezult[0] === undefined) {
+      item = { ...state };
+    } else {
+      item = { ...rezult[rezult.length - 1] };
+    };
```

Comment:
```suggestion
  let item = { ...state };

  for (const action of actions) {
```
просто створи копію стейта ПОЗА межами циклу. І ці іфи тобі не потрібні. Також краще використовуй const action of actions. Це більш інтуітивно. `i` - це наштовзує на думку, що тут індекс. В нас же індексів тут немає